Sources/Onboarding/Shields/AntiAdBlockWarningView.swift

```diff
@@ -0,0 +1,142 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveUI
+import DesignSystem
+import Strings
+import Preferences
+import BraveShields
+import SnapKit
+import Data
+
+public struct AntiAdBlockWarningView: View {
+  public typealias DismissCallback = (Bool) -> Void
+  
+  let url: URL
+  let dismiss: DismissCallback
+  
+  private var playerDescription: AttributedString? {
+    do {
+      return try AttributedString(markdown: Strings.Shields.antiAdBlockWarningBravePlayerDescription)
+    } catch {
+      assertionFailure()
+      return nil
+    }
+  }
+  
+  public var body: some View {
+    ScrollView {
+      VStack(alignment: .leading, spacing: 24) {
+        Text(Strings.Shields.antiAdBlockWarningTitle)
+          .font(.headline).foregroundStyle(Color(braveSystemName: .textPrimary))
+        Text(Strings.Shields.antiAdBlockWarningDescription)
+        Text(Strings.Shields.antiAdBlockWarningDescription2)
+        
+        if let playerDescription = playerDescription {
+          VStack(spacing: 0) {
+            Image("youtube_warning_address_bar", bundle: .module)
+            Text(playerDescription)
+              .multilineTextAlignment(.center)
+              .font(.caption)
+              .padding()
+              .frame(maxWidth: .infinity, alignment: .center)
+          }
+          .clipShape(RoundedRectangle(cornerRadius: 8))
+          .overlay(
+            RoundedRectangle(cornerRadius: 8)
+              .stroke(Color(braveSystemName: .gray20), lineWidth: 1)
+          )
+        }
+        
+        VStack(alignment: .center) {
+          Button {
+            ShieldPreferences.hasSeenAntiAdBlockWarning.value = true
+            Domain.setBraveShield(forUrl: url, shield: .AllOff, isOn: true, isPrivateBrowsing: false)
+            dismiss(true)
+          } label: {
+            Text(Strings.Shields.antiAdBlockWarningConfirmationButton)
+              .multilineTextAlignment(.center)
```

Comment:
all Text inside this VStack seem to have multiline center alignment, remove from each individual text and add to containing VStack instead